lib/phx_tools_web/live/phx_tools_live/index.ex

```diff
@@ -17,16 +17,29 @@ defmodule PhxToolsWeb.PhxToolsLive.Index do
     do: assign(socket, :operating_system, operating_system)
 
   @impl Phoenix.LiveView
-  def handle_params(_params, _uri, socket),
-    do: {:noreply, apply_action(socket, socket.assigns.live_action)}
+  def handle_params(_params, _uri, socket) do
+    apply_action(socket, socket.assigns.live_action)
+  end
+
+  defp apply_action(socket, :index) do
+    case socket.assigns.operating_system do
+      "Linux" -> {:noreply, assign(socket, :live_action, :linux)}
+      "Mac" -> {:noreply, assign(socket, :live_action, :macOS)}
+      _other -> {:noreply, socket}
+    end
+  end
 
   defp apply_action(socket, :linux) do
-    assign(socket, seo_attributes: %{url: url(~p"/linux")})
+    {:noreply,
+     socket
+     |> assign(:live_action, :linux)
+     |> assign(seo_attributes: %{url: url(~p"/linux")})}
   end
 
   defp apply_action(socket, :macOS) do
```

Comment:
Can you combine these two so you don't have to hardcode assigning seo attributes, but depend on the live_action passed to it instead?